coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -2037,7 +2037,18 @@ func InsertWorkspaceResource(ctx context.Context, db database.Store, jobID uuid.
 				Health:               health,
 				DisplayOrder:         int32(app.Order),
 				Hidden:               app.Hidden,
-				OpenIn:               app.OpenIn,
+				OpenIn: func() database.WorkspaceAppOpenIn {
+					switch app.OpenIn {
+					case sdkproto.AppOpenIn_WINDOW:
+						return database.WorkspaceAppOpenInWindow
+					case sdkproto.AppOpenIn_SLIM_WINDOW:
+						return database.WorkspaceAppOpenInSlimWindow
+					case sdkproto.AppOpenIn_TAB:
+						return database.WorkspaceAppOpenInTab
+					default:
+						return database.WorkspaceAppOpenInSlimWindow
+					}
+				}(),
```

Comment:
That's an interesting form :) Perhaps move it above the `db.InsertWorkspaceApp` in the same way as `sharingLevel`?